js/transitions.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Handler to await completion of `CSSTransitions` to be run at point of execution.
+ * An optional `transition-property` to await can be specified, else all properties will be included.
+ * @param {jQuery} $element
+ * @param {String} property
+ * @returns {Promise}
+ */
+export async function transitionsEnded($element, property = null) {
+  return new Promise(resolve => {
+    if (!($element instanceof $)) $element = $($element);
+    // only listen to transitions running after execution
+    const onRun = () => $element.on('transitioncancel transitionend', onEnd);
+    const onEnd = () => {
+      if (!hasActiveTransition($element, property)) done(true);
+    };
+    const done = (didTransition) => {
+      $element
+        .off('transitionrun', onRun)
+        .off('transitioncancel transitionend', onEnd);
+      resolve(didTransition);
+    };
+    $element.on('transitionrun', onRun);
+    if (!willTransition($element, property)) done(false);
+  });
+}
+
+/**
+ * Returns all `CSSTransitions` affecting an element.
+ * @param {jQuery} $element
+ * @returns {Array<CSSTransition>}
+ */
+export function getTransitions($element) {
+  let element = $element;
+  if ($element instanceof $) element = element[0];
+  return element.getAnimations().filter(animation => animation instanceof CSSTransition);
```

Comment:
This is nice, I hadn't realised the browser api kept a list of active transitions.